### ISSUE TITLE
slow motion in "Barbarian" (Sound is buggy)

### ISSUE THREAD

**Issue by FloLore** [ColinPitrat/caprice32#117]
Hi ColinPitrat,

It seems that there is a problem with "Barbarian" from Palace Software, the game seems to go faster than normal which causes a bad vsync, the sound is choppy, bug. When I pause the game, the sound works very well (continuous music).

I searched if a different configuration of "cap32.cfg" could solve this problem but without finding ... Could you tell me which option I wrongly configured? If the problem comes from the emulator, could you fix it?

Best regards,

**Comment by ColinPitrat** [comment]
Hi,

I couldn't reproduce the issue (but I couldn't leave the demo game of the game either, not sure how to start playing). The result I have is similar to what I saw in this video: https://www.youtube.com/watch?v=8lCRfpPIVBI

Is the emulator running at 100% speed ? (this should be visible at the top-left corner of the screen, if it's not you can have it by pressing F8).

Regards,
Colin

**Comment by FloLore** [comment]
Indeed, with the F8 key the game runs at 28 fps and 50% of its speed ... More precisely, when loading it is Indeed, with the F8 key the game runs at 28 fps and 50% of its speed ... More precisely, when loading it is 100%, but once the demonstration mode started, the emulator drops to 28% and keeping around ...
What a shame, I like this game! I tried the original version, and cracked, downloadable on "http://www.cpc-power.com/index.php?page=detail&onglet=dumps&num=40", but no version works correctly with Caprice32, even with the default configuration file.
If you can not reproduce this problem, it is that the problem comes from my PC but where (If you can advise me) ? In addition, my other games work very well with your emulator (that I like).

To be able to play and leave the "Demo mode", press the "Enter" key to change the mode, then "Space" to select mode 1 (Vs CPU, Joystick), 2 (Vs CPU, Keyboard), or 3 (PVP)

**Comment by ColinPitrat** [comment]
The reason it works for me and not for you could be:
 - that I have a faster computer (*)
 - that something in your setup makes it slower

I checked a profile of caprice while the game was running and it doesn't seem odd which, I think, eliminates the first possibility. What setup do you use (OS, processor, memory, sound card, video card, cap32.cfg ...) ? What other setups did you try ?
One thing you can do is double check you're building in release mode (this should be the case unless you explicitly built with `make debug` in which case you should have a `.debug` file in the build directory).
Then you can try changing some config parameters: different video drivers, disabling sound, etc ... The emulation speed can help you determine if it helps or not.


(*) This doesn't mean that your computer is not powerful enough though (especially if other games work fine). The game may make use of some functionality that are making the emulator slow, which would be a bug that require fixing.

**Comment by FloLore** [comment]
Here are the details of my pc: i7 6850k, asus motherboard, 32gb ram, asus 1080ti and a ZxR creative sound card. My OS is windows 10 pro.

When you first launch an application under w10, the OS automatically registers compatibility values ​​in the registry, recently I made some modifications at this level for caprice32. Could you give me your values ​​(only for the application cap32.exe ") to these keys:" [HKEY_CURRENT_USER \ Software \ Microsoft \ Windows \ NT \ CurrentVersion \ AppCompatFlags \ Layers] "and" [HKEY_LOCAL_MACHINE \ SOFTWARE \ Microsoft \ Windows NT \ CurrentVersion \ AppCompatFlags \ Layers]. Maybe the problem comes from there ...

As you make a "release" of your emulator, I do not feel the need to compile the sources myself, so I use directly your latest version updated and compiled on your site: "https://github.com/ ColinPitrat / caprice32 / releases / download / latest / cap32-win64.zip "

But I will try to compiled with the sources to have access to the mode "debug", I will keep you informed of the rest ... Thank you Colin!

**Comment by ColinPitrat** [comment]
Interesting, when you know a version that works and another that doesn't, a nice way to find where the problem comes from is to do a git bisect.
The message you mention was added by: https://github.com/ColinPitrat/caprice32/commit/fed782e5a05180a90049360f60755f356d79634d
but I don't think this is the source of the issue because it really just adds a comparison, some assignments and the message itself.

Are you able to build clone from git and build ? To start the `git bisect`, you need to know the commit of the version that works. You can view the history with `git log` and checkout a particular commit with `git checkout <commit sha1>` where `<commit sha1>` is the commit identifier, looking like `a7e51b036904b46794e8f70d590e9f594adcd0d4` for example.

Then the procedure for a git bisect is the following:
```
git bisect start
git bisect bad  # Latest version is not working
git bisect good a7e51b036904b46794e8f70d590e9f594adcd0d4  # Replace sha1 with the known working version
```
Then git will automatically checkout the version that is mid-way between those two. You have to build it, try it and see if it works. If it does, run `git bisect good` otherwise run `git bisect bad`. Reiterate until git tells you (after a few steps) `<commit sha1> is the first bad commit`. Congratulations, you've found the source of the issue :-)
More details: https://git-scm.com/docs/git-bisect

For the registry, I can't really help you because I don't have windows.

**Comment by FloLore** [comment]
I did the research with bisect, that gives me this result: https://ibb.co/1T9mkJc

**Comment by ColinPitrat** [comment]
Actually, I think I do observe a difference between the bad & good versions (the max speed is much lower when I don't limit it) so I can try to do the bisect myself.

**Comment by ColinPitrat** [comment]
In fact, the difference I observed was because the first build I looked at was in debug with profiling instrumentation (so much slower). When building in release, both versions have similar performances.
There are a lot of log messages in the console, which may slow down if your console is slow. I'm trying to remove them (they are useless anyway), tell me if it helps (this should be available as soon as appveyor finishes its build).
